packages/harpguru-core/src/utils/determine-matrix-dimensions/determine-matrix-dimensions.ts

```diff
@@ -7,9 +7,10 @@ export const determineMatrixDimensions = (
   matrix: ReadonlyArray<ReadonlyArray<unknown>>
 ): MatrixDimensions => {
   const { length: rows } = matrix
-  const {
-    [0]: { length: columns },
-  } = matrix
+  const columns = (() => {
```

Comment:
This is necessitated by some basic assumption about the empty harpface2 matrix. I should review whether this is necessary or the most semantic solution.

Reply:
It's fine. Clear enough and functional.